### PR TITLE
adding sample pipeline

### PR DESCRIPTION
**Pull request by Javidleo**
## Summary
adding sample request pipe line with delegates.

## Changes
- adding Exception Handler
- adding Authorization Handler
- adding CORS Handler
- Prevent Ip's Starting with 192.168 (Iran IP example) to Use Service.

### CODE REVIEW

File: src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Context.cs
Comment (mohammadKarimi) [inline src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Context.cs:1]: for 1 or 2 property, it is better to use primary constructor.
Record :) like.

File: src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Exceptions/ForbiddenAccessException.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿namespace PipelineDesignPattern.Exceptions;
+
+public class ForbiddenAccessException(string message) : Exception(message)
```
Comment (mohammadKarimi) [inline src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Exceptions/ForbiddenAccessException.cs:3]: Custom Exception 👍

File: src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Pipe.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        try
+        {
+            "ExceptionHandling Started".Dump();
+
+            action(context);
+
+            "ExceptionHandling End".Dump();
+        }
+        catch (Exception ex)
```
Comment (mohammadKarimi) [inline src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Pipe.cs:18]: this line catches all type of Exception 👎 
you should develop issue related exception

File: src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Program.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        Console.WriteLine("Please Enter your Ip");
+        Ip = Console.ReadLine()!;
+    }
+    else
+        break;
+}
+
+Context httpContext = new() { Ip = Ip };
+var pipe = new Pipe();
```
Comment (mohammadKarimi) [inline src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/Program.cs:19]: Target Type Expression is better. ( I Like it) but you can not to use it.

File: src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        {"111.111.111.111", new User("ali", "rezaie") },
+        {"222.333.333.333", new User("mohammad", "ahmadi") },
+        {"444.444.444.444", new User("reza", "navidi") },
+
+    };
+    public void GetUserData(Context context)
+    {
+        var user = _users[context.Ip];
+        if (user == null)
```
Comment (mohammadKarimi) [inline src/Javid Hassani/PipelineDesignPattern/PipelineDesignPattern/UserService.cs:19]: it is better to use constant matching instead == operator

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]

**Comment by Javidleo** [comment]
Thanks for review. I will make changes then notify you to review it again.

**Comment by Javidleo** [comment]
@mohammadKarimi  @thisisnabi  
new changes as you requested.

**Review by mohammadKarimi — APPROVED** [review]
Perfect

**Pull request merged**